react-events-and-state/src/index.jsx

```diff
@@ -0,0 +1,25 @@
+import React from 'react';
+import ReactDOM from 'react-dom/client';
+
+class CustomButton extends React.Component {
+  constructor(props) {
+    super(props);
+    this.handleClick = this.handleClick.bind(this);
+    this.state = { isClicked: false };
+  }
+
+  handleClick() {
+    this.setState({ isClicked: true });
+    this.props.text = 'Thank you!';
```

Comment:
`this.props` should be read only, we never directly set `this.props` from inside the component.